### PR TITLE
fix(#4190): useLabels/useLabel consider reordering IDREFs so that aria-label comes before aria-labelledby

### LINKED ISSUE #4190: useLabels/useLabel consider reordering IDREFs so that aria-label comes before aria-labelledby
In `useLabels`:

https://github.com/adobe/react-spectrum/blob/11ca4ea9de96fbf052da1edb13c1ba777719ea4e/packages/%40react-aria/utils/src/useLabels.ts#L32

and `useLabel`: 

https://github.com/adobe/react-spectrum/blob/11ca4ea9de96fbf052da1edb13c1ba777719ea4e/packages/%40react-aria/label/src/useLabel.ts#L50

`aria-labelledby` IDREFs are concatenated before the IDREF for the `label` or `aria-label`. This oftentimes means that the more specific label for the input element follows the less specific label for the group. This is different from the behavior of controls contained within a `fieldset` element, in which case the more specific label for the `input` will be announced before the name of the `fieldset` that contains it.

We see the effects of this ordering in controls like DatePicker, DateRangePicker and RangeSlider. In [DateRangePicker](https://react-spectrum.adobe.com/react-spectrum/DateRangePicker.html#non-contiguous-ranges), for example, each date segment starts with the name of the group, rather than the more specific name of the segment, as follows:

- Empty, Time off request Start Date month, stepper, Time of request group
- Empty, Time off request Start Date day, stepper
- Empty, Time off request Start Date year, stepper
- Empty, Time off request End Date month, stepper
- Empty, Time off request End Date day, stepper
- Empty, Time off request End Date year, stepper
- Time off request Calendar, dialog popup collapsed, button

A behavior that favors the more specific input over the group might announce as follows:

- Empty, month, Start Date, Time off request, stepper, Time of request group
- Empty, day, Start Date, Time off request, stepper
- Empty, year, Start Date, Time off request, stepper
- Empty, month, End Date, Time off request, stepper
- Empty, day, End Date, Time off request, stepper
- Empty, year, End Date, Time off request, stepper
- Calendar, Time off request, dialog popup collapsed, button

This issue received comment in a Twitter thread related to the DatePicker release: https://twitter.com/aardrian/status/1539730646087499782. 

In order to ensure that one can disambiguate between multiple instances of the same control within an web application, when a screen reader brings up a list of form elements using VO + U or NVDA + F7 for example, I'm not sure we can forego using `aria-labelledby` to include the group label as part of the input label in favor of using the input label alone within a fieldset. But we can put the text that is more likely to change when navigating through controls first.

### PR DESCRIPTION
**Pull request by majornista**
Closes #4190 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 4190](https://github.com/adobe/react-spectrum/issues/4190).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

Adobe/Accessibility

### CODE REVIEW

File: packages/@react-aria/datepicker/src/useDateSegment.ts [hunk 1]
```diff
@@ -346,7 +346,7 @@ export function useDateSegment(segment: DateSegment, state: DateFieldState, ref:
   // This is needed because VoiceOver on iOS does not announce groups.
   let name = segment.type === 'literal' ? '' : displayNames.of(segment.type);
   let labelProps = useLabels({
-    'aria-label': (ariaLabel ? ariaLabel + ' ' : '') + name,
+    'aria-label': `${name}${ariaLabel ? `, ${ariaLabel}` : ''}${ariaLabelledBy ? ', ' : ''}`,
```
Comment (snowystinger) [inline packages/@react-aria/datepicker/src/useDateSegment.ts:349]: commas are being added to get the screen reader to make a small pause?

File: packages/@react-spectrum/datepicker/test/DatePicker.test.js [hunk 2]
```diff
@@ -431,7 +431,7 @@ describe('DatePicker', function () {
       );
 
       let combobox = getAllByRole('group')[0];
-      expect(getTextValue(combobox)).toBe('2/3/2019, 8:45 AM');
+      expect(getTextValue(combobox).replaceAll(' ', ' ')).toBe('2/3/2019, 8:45 AM');
```
[the remark this answers is not in the document]
Comment (majornista) [reply packages/@react-spectrum/datepicker/test/DatePicker.test.js:434]: Different versions of node are adding, or not adding, non-breaking spaces in the date formatting.

File: packages/@react-aria/datepicker/src/useDateSegment.ts (hunk 1 above)
Comment (majornista) [reply packages/@react-aria/datepicker/src/useDateSegment.ts:349]: Yes. Particularly in DateRangePicker, the labels become a  something of a string of words. For example, "5 - May, Date range Start date month, stepper." With the changes suggested in this PR, the accessibility name will read from the value most likely to change to the value least likely to change as the user navigates between segments. "5 - May, month, Start date, Date range, stepper" The commas help make more sense of the reordered string.

File: packages/@react-spectrum/datepicker/test/DatePicker.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/datepicker/test/DatePicker.test.js:434]: ah, lets just match the value from the version of node that we run CI in, currently 16.20

**Review by snowystinger — APPROVED** [review]

**Comment by majornista** [comment]
@snowystinger Thanks for the commit, removing my over-engineering in the tests, and for your approval.

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**